### PR TITLE
[Sema] Test cases to avoid misleading case will never be executed war…

### PR DESCRIPTION
**Pull request by jreference**
…nings

EDIT: #50745 is an existing issue, it was not fixed previously. This PR simply adds a test case to avoid regressions with a FIXME note.

### CODE REVIEW

**Review by xedin — COMMENTED** [review]
@jreference Could you please move the test-case to `test/Constraints/patterns.swift` and reference issue in a comment similar to i.e. https://github.com/apple/swift/blob/main/test/Constraints/diagnostics.swift#L1415?

File: test/Sema/inaccurate-warnings-in-single-case-switch.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// RUN: %target-typecheck-verify-swift
```
Comment (AnthonyLatsis) [inline test/Sema/inaccurate-warnings-in-single-case-switch.swift:1]: This is a [SIL diagnostic](https://github.com/apple/swift/blob/8ab43a4a444702eb96e60ef78e4512947b16dbbf/lib/SILGen/SILGenPattern.cpp#L1069). Running semantic analysis won’t catch a regression.

Comment (xedin) [reply test/Sema/inaccurate-warnings-in-single-case-switch.swift:1]: Oh! Yeah, that needs to be moved to test/SILGen then!

**Comment by AnthonyLatsis** [comment]
Please see https://github.com/apple/swift/issues/50745#issuecomment-1542926973 before this gets approved.

**Comment by AnthonyLatsis** [comment]
What happened?

**Comment by AnthonyLatsis** [comment]
> On a sidenote, is there a simple way to tell whether a warning is being emitted in the semantic analysis phase or in the SILGen phase?

#### From a user perspective

* if you see it in `-parse` mode, then it was emitted during parsing.
* if you see it in `-typecheck` mode but not in `-parse` mode, then it was emitted during Sema.
* if you see it in `-emit-silgen` mode but not in `-typecheck` mode, then it was emitted during SILGen.
* And so on.

#### From a compiler developer perspective

* Compile with the frontend option `-debug-diagnostic-names`.
* Search the codebase for the identifier printed next to the warning.

**Pull request closed without merging**